includes/service-providers/constant_contact/class-newspack-newsletters-constant-contact.php

```diff
@@ -887,6 +887,71 @@ public function get_contact_data( $email, $return_details = false ) {
 		return get_object_vars( $contact );
 	}
 
+	/**
+	 * Get the lists a contact is subscribed to.
+	 *
+	 * @param string $email The contact email.
+	 *
+	 * @return string[] Contact subscribed lists IDs.
+	 */
+	public function get_contact_lists( $email ) {
+		$contact_lists = [];
+		$contact_data = self::get_contact_data( $email );
+		if ( is_wp_error( $contact_data ) ) {
+			return $contact_lists;
+		}
+		if ( ! empty( $contact_data['list_memberships'] ) ) {
+			$contact_lists = array_merge( $contact_lists, $contact_data['list_memberships'] );
+		}
+		if ( ! empty( $contact_data['taggings'] ) ) {
+			$contact_lists = array_merge( $contact_lists, $contact_data['taggings'] );
+		}
+
+		return $contact_lists;
+	}
+
+	/**
+	 * Update a contact lists subscription.
+	 *
+	 * @param string   $email           Contact email address.
+	 * @param string[] $lists_to_add    Array of list IDs to subscribe the contact to.
+	 * @param string[] $lists_to_remove Array of list IDs to remove the contact from.
+	 *
+	 * @return true|WP_Error True if the contact was updated or error.
+	 */
+	public function update_contact_lists( $email, $lists_to_add = [], $lists_to_remove = [] ) {
+		$cc           = new Newspack_Newsletters_Constant_Contact_SDK( $this->api_key(), $this->api_secret(), $this->access_token() );
+		$contact_data = $this->get_contact_data( $email );
+		if ( is_wp_error( $contact_data ) ) {
+			/** Create contact */
+			// Call Newspack_Newsletters_Subscription's method (not the provider's directly),
+			// so the appropriate hooks are called.
+			$contact_data = Newspack_Newsletters_Subscription::add_contact( [ 'email' => $email ] );
+			if ( is_wp_error( $contact_data ) ) {
+				return $contact_data;
+			}
+		}
+
+		// Remove lists or tags from contact.
+		if ( $lists_to_remove ) {
```

Comment:
Does this need to be `if ( empty( $lists_to_remove ) ) {` instead? I believe an empty array will always evaluate to `true`, so this might result in extra requests that don't end up doing anything, right?

Reply:
Empty arrays evaluate to `false`, unless this changed in some php versions.

But I'm not against a more explicit check. Explicit > Implicit :)

Reply:
Ah, you're right! I must have been thinking of JS. I'm okay with it either way